transformations/transformations/global_var_offload.py

```diff
@@ -0,0 +1,275 @@
+        """
+
+        update_device = ()
+        update_host = ()
+
+        # build offload pragmas
+        if self._acc_copyin:
+            update_device += as_tuple(Pragma(keyword='acc',
+                                             content='update device(' + ','.join(self._acc_copyin) + ')'),)
+        if self._enter_data_copyin:
+            update_device += as_tuple(Pragma(keyword='acc',
+                                             content='enter data copyin(' + ','.join(self._enter_data_copyin) + ')'),)
+        if self._enter_data_create:
+            update_device += as_tuple(Pragma(keyword='acc',
+                                             content='enter data create(' + ','.join(self._enter_data_create) + ')'),)
+        if self._exit_data:
+            update_host += as_tuple(Pragma(keyword='acc',
+                                           content='exit data copyout(' + ','.join(self._exit_data) + ')'),)
+        if self._acc_copyout:
+            update_host += as_tuple(Pragma(keyword='acc', content='update self(' + ','.join(self._acc_copyout) + ')'),)
+
+        # replace Loki pragmas with acc data/update pragmas
+        pragma_map = {}
+        for pragma in FindNodes(Pragma).visit(routine.body):
+            if pragma.keyword == 'loki':
+                if 'update_device' in pragma.content:
+                    if update_device:
+                        pragma_map[pragma] = update_device
+                    else:
+                        pragma_map[pragma] = None
+
+                if 'update_host' in pragma.content:
+                    if update_host:
+                        pragma_map[pragma] = update_host
+                    else:
+                        pragma_map[pragma] = None
+
+        routine.body = Transformer(pragma_map).visit(routine.body)
+
+        # build new imports to add offloaded global vars to driver symbol table
+        new_import_map = {}
+        for s in self._var_set:
+            if s in routine.symbol_map:
+                continue
+
+            if new_import_map.get(self._modules[s], None):
+                new_import_map[self._modules[s]] += as_tuple(s)
+            else:
+                new_import_map.update({self._modules[s]: as_tuple(s)})
+
+        new_imports = ()
+        for k, v in new_import_map.items():
+            new_imports += as_tuple(Import(k, symbols=tuple(Variable(name=s, scope=routine) for s in v)))
+
+        # add new imports to driver subroutine sepcification
+        import_pos = 0
+        if (old_imports := FindNodes(Import).visit(routine.spec)):
+            import_pos = routine.spec.body.index(old_imports[-1]) + 1
+        if new_imports:
+            routine.spec.insert(import_pos, Comment(text=
+               '![Loki::GlobalVarOffload].....Adding global variables to driver symbol table for offload instructions'))
+        import_pos += 1
+        for index, i in enumerate(new_imports):
+            routine.spec.insert(import_pos + index, i)
```

Comment:
`Section.insert` accepts also a tuple of nodes, so could just be `routine.spec.insert(import_pos, new_imports)`

Reply:
Ah yes, that's much cleaner, thanks for the suggestion!